### PR TITLE
Implement search input

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Implement search input

image

storybook: https://6684faab2d0c99a3e93bf145-dnwjfebfzt.chromatic.com/

### CODE REVIEW

File: src/components/SearchInput/SearchInput.stories.tsx [hunk 1]
```diff
@@ -9,5 +9,5 @@ export default {
 type Story = StoryObj<typeof SearchInput>;
 
 export const Default: Story = {
-  args: {},
+  args: { placeholder: 'Search', inputType: 'input', onChange: () => {} },
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/SearchInput.stories.tsx]: enter pls.

Comment (hungdo-agilityio) [inline src/components/SearchInput/SearchInput.stories.tsx]: onChange is empty function is not correct, please read this https://storybook.js.org/docs/essentials/actions

File: src/components/SearchInput/searchInput.module.css [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  border-radius: var(--border-radius);
+  width: 100%;
+}
+
+.searchInput::placeholder {
+  color: var(--color-disable);
+  font-size: var(--font-size-md);
+}
+
+.searchIcon {
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/searchInput.module.css:19]: You could use pseudo-elements like `::before` to reduce DOM here.

File: src/components/SearchInput/SearchInput.tsx [hunk 3]
```diff
@@ -1,11 +1,28 @@
...
+  onChange,
+}: SearchInputProps) => (
+  <form className={styles.container}>
+    <input
+      type={inputType}
+      placeholder={placeholder}
+      onChange={onChange}
+      className={styles.searchInput}
+    />
+    <div className={styles.searchIcon}>
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/SearchInput.tsx:22]: You could use pseudo-elements like ::before to reduce DOM here.

File: src/components/SearchInput/searchInput.module.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
+.container {
+  position: relative;
+  display: inline-block;
+}
+
+.searchInput {
+  padding: 0.75rem 1rem;
```
Comment (hungdo-agilityio) [inline src/components/SearchInput/searchInput.module.css]: What happen if I type the long sentence, will the icon overlap the text?

File: src/app.module.css [hunk 5]
```diff
@@ -0,0 +1,14 @@
+/* example style */
+
+.right {
+  float: right;
+}
+
+.position {
```
Comment (thanhnguyen-agilityio) [inline src/app.module.css]: `absolute-center`

File: src/components/SearchInput/SearchInput.tsx [hunk 6]
```diff
@@ -1,11 +1,31 @@
...
-  return <></>;
-};
+const Input = ({
+  placeholder,
+  inputType,
+  onChange,
+  children,
+  positionIcon,
+}: InputProps) => (
+  <form className={styles.container}>
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/SearchInput.tsx]: not a form, a div instead

File: src/components/SearchInput/searchInput.module.css [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+.searchInput::placeholder {
+  color: var(--color-disable);
+  font-size: var(--font-size-md);
+}
+
+.searchIcon {
+  color: var(--color-text-product);
+  position: absolute;
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/searchInput.module.css:21]: is it duplicate with `.absolute-cênter` in app module css

File: src/components/SearchInput/searchInput.module.css [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  top: 50%;
+  right: 1rem;
+  transform: translateY(-50%);
+  cursor: pointer;
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/searchInput.module.css]: add `.pointer` class in general to use.

File: src/components/SearchInput/searchInput.module.css [hunk 9]
```diff
@@ -0,0 +1,26 @@
+.container {
+  position: relative;
```
Comment (thanhnguyen-agilityio) [inline src/components/SearchInput/searchInput.module.css]: add `.relative` generic  class to use

File: src/components/SearchInput/SearchInput.stories.tsx (hunk 1 above)
Comment (thanhtrantruong47) [reply src/components/SearchInput/SearchInput.stories.tsx]: Fixed

Comment (thanhtrantruong47) [reply src/components/SearchInput/SearchInput.stories.tsx]: Fixed

File: src/components/SearchInput/searchInput.module.css (hunk 4 above)
Comment (thanhtrantruong47) [reply src/components/SearchInput/searchInput.module.css]: Fixed I define maxlenght 20

File: src/app.module.css (hunk 5 above)
Comment (thanhtrantruong47) [reply src/app.module.css]: Fixed

File: src/components/SearchInput/SearchInput.tsx (hunk 6 above)
Comment (thanhtrantruong47) [reply src/components/SearchInput/SearchInput.tsx]: Fixed

File: src/components/SearchInput/searchInput.module.css (hunk 8 above)
Comment (thanhtrantruong47) [reply src/components/SearchInput/searchInput.module.css]: Fixed

File: src/components/SearchInput/searchInput.module.css (hunk 9 above)
Comment (thanhtrantruong47) [reply src/components/SearchInput/searchInput.module.css]: Fixed

File: src/components/Input/Input.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+import styles from './input.module.css';
+import utils from '../../styles/modules/utils.module.css';
+import EndIcon from '../Icon/Icon';
+
+interface InputProps {
+  placeholder: string;
+  inputType: string;
+  onChange: () => void;
+  children?: ReactNode;
+}
```
Comment (hungdo-agilityio) [inline src/components/Input/Input.tsx]: These can be extends from props of input html

File: src/components/Input/Input.tsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
+import { ReactNode } from 'react';
+import styles from './input.module.css';
+import utils from '../../styles/modules/utils.module.css';
+import EndIcon from '../Icon/Icon';
+
+interface InputProps {
+  placeholder: string;
+  inputType: string;
+  onChange: () => void;
+  children?: ReactNode;
```
Comment (hungdo-agilityio) [inline src/components/Input/Input.tsx]: should be `endIcon`

File: src/components/Input/Input.tsx (hunk 10 above)
Comment (thanhtrantruong47) [reply src/components/Input/Input.tsx]: Fixed

File: src/components/Input/Input.tsx (hunk 11 above)
Comment (thanhtrantruong47) [reply src/components/Input/Input.tsx]: Fixed

**Review by thanhnguyen-agilityio — APPROVED** [review]

File: src/components/Input/Input.tsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
+import { InputHTMLAttributes, ReactNode } from 'react';
+import styles from './input.module.css';
+import utils from '../../styles/modules/utils.module.css';
+import EndIcon from '../Icon/EndIcon';
+
+interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
+  onChange: () => void;
```
Comment (hungdo-agilityio) [inline src/components/Input/Input.tsx:7]: onChange is already include in `InputHTMLAttributes<HTMLInputElement>`

**Pull request merged**